glean/rts/factset.cpp

```diff
@@ -245,12 +245,11 @@ Id FactSet::define(Pid type, Fact::Clause clause, Id) {
     error("key too large: {}", clause.key_size);
   }
   const auto next_id = firstFreeId();
-  auto fact = Fact::create({next_id, type, clause});
+  auto fact = facts.alloc(next_id, type, clause);
```

Comment:
Why is another abstraction over Fact::create is needed?

Reply:
See the PR description, the entire point of this is to put an abstraction layer over the fact table so that we can replace it more easily.